### DISCUSSION TITLE
目前的 thumbfast.lua 有办法在串流流媒体的时候正常的生成缩略图吗？

### DISCUSSION THREAD

**Discussion by wathper** [dyphire/mpv-config#63] [General]
我查看了上游的 thumbfast.lua 仓库 https://github.com/po5/thumbfast/issues/73
有人提到这个脚本目前只能在 youtube 少数流媒体上生成缩略图。有其他的解决办法使它能在其他的网站串流时也正常生成缩略图吗？

**Comment by dyphire** [comment]
那要看你对正常的定义了，在我看来所有的串流协议都无法正常工作，因为生成缩略图的速度太慢了，这么低的效率是不可用的。
如果你尝试挂载 smb/webdav/网盘 然后用 mpv 播放就会发现 thumbfast.lua 不会识别为串流协议而禁用缩略图从而导致奇慢的缩略图生成过程。我一直在考虑移除该缩略图脚本，因为它对 dovi 和挂载文件只产生副作用

唯一可行的缩略图实行方案就是 mpv 上游原生提供一个管道用于缩略图生成接口，这也是其他支持缩略图的播放器的做法

**Comment by dyphire** [comment]
在提交 https://github.com/dyphire/mpv-config/commit/13d12fd1addbf6d81419e83927484698ac75cf48 中删除了完全无用的色调映射功能（低到令人发指的缩略图生成速度），并为杜比视界配置5的视频禁用不正常缩略图

**Reply by wathper** [reply]
我回看缩略图时一般是视频播放了一段时间后进行回顾，因为已经渲染好了，对生成速度慢这点到是没有太过注意到。
不过刚测试了一下将 PikPak 挂载到本地用mpv播放确实有点慢了。
